controllers/dscinitialization/dscinitialization_controller.go

```diff
@@ -124,8 +124,12 @@ func (r *DSCInitializationReconciler) Reconcile(ctx context.Context, req ctrl.Re
 		}
 	} else {
 		r.Log.Info("Finalization DSCInitialization start deleting instance", "name", instance.Name, "finalizer", finalizerName)
-		if err := r.removeServiceMesh(instance); err != nil {
-			return reconcile.Result{}, err
+		// when delete DSCI, we only delete SMCP if servicemesh.management is Managed
+		// if it is unamanged or removed, we do not explicted remove it: either we should keep it or nothinig to delete
+		if instance.Spec.ServiceMesh.ManagementState == operatorv1.Managed {
```

Comment:
IMHO, regardless of the management state, we should _not_ remove. So, I think we should always skip `removeServiceMesh`.

The right expectation is that, all circumstances, the operator should _never_ delete the SMCP.